netdef/main.go

```diff
@@ -23,16 +30,28 @@ func main() {
 			if err != nil {
 				return err
 			}
-			defer fi.Close()
 
-			var cfg netdef.Config
-			if err := json.NewDecoder(fi).Decode(&cfg); err != nil {
+			cfg := &netdef.Config{}
+			if err = json.NewDecoder(fi).Decode(cfg); err != nil {
+				fi.Close()
 				return err
 			}
+			fi.Close()
 
-			if err := netdef.Create(&cfg); err != nil {
+			r, err := cfg.Create()
+			if err != nil {
+				return err
+			}
+
+			fi, err = os.Open(c.String("output"))
```

Comment:
I would maybe encapsulate this logic in a separate function, that way we don't have to reuse the variable names

Reply:
can do